File: 프로그래머스/2/42860. 조이스틱/조이스틱.py
def solution(name):
    cursor_num = len(name) - 1  
    spell_num = 0
    
    for i, s in enumerate(name):
        spell_num += min(ord(s) - ord('A'), ord('Z') - ord(s) + 1)

        next = i + 1
        while next < len(name) and name[next] == 'A': next += 1

        cursor_num = min([cursor_num, 2 * i + len(name) - next, i + 2 * (len(name) - next)])
            
    return spell_num + cursor_num

File: 프로그래머스/2/42860. 조이스틱/조이스틱.js
function solution(name) {
    let cursorNum = name.length - 1;
    let spellNum = 0;
    
    for (let i = 0; i < name.length; i++) {
        const s = name[i];
        spellNum += Math.min(s.charCodeAt(0) - 'A'.charCodeAt(0), 'Z'.charCodeAt(0) - s.charCodeAt(0) + 1);

        let next = i + 1;
        while (next < name.length && name[next] === 'A') next++;

        cursorNum = Math.min(cursorNum, 2 * i + name.length - next, i + 2 * (name.length - next));
    }
    
    return spellNum + cursorNum;
}
